pkg/disruptors/controller.go

```diff
@@ -40,29 +39,28 @@ func (c *PodController) Visit(ctx context.Context, visitor PodVisitor) error {
 	defer cancelVisit()
 
 	// make space to prevent blocking go routines
-	errCh := make(chan error, len(c.targets))
+	doneCh := make(chan error, len(c.targets))
 
-	wg := sync.WaitGroup{}
 	for _, pod := range c.targets {
-		wg.Add(1)
 		go func(pod corev1.Pod) {
-			if err := visitor.Visit(visitCtx, pod); err != nil {
-				errCh <- err
-			}
-
-			wg.Done()
+			doneCh <- visitor.Visit(visitCtx, pod)
 		}(pod)
 	}
 
-	wg.Wait()
-
-	select {
-	case e := <-errCh:
-		return e
-	case <-ctx.Done():
-		return ctx.Err()
-	default:
-		return nil
+	pending := len(c.targets)
+	for {
+		select {
+		case e := <-doneCh:
+			if e != nil {
+				return e
+			}
+			pending--
+			if pending == 0 {
+				return nil
+			}
+		case <-ctx.Done():
+			return ctx.Err()
+		}
 	}
```

Comment:
I think a regular, c-style for loop migth be slightly more readable. WDYT?

```suggestion
	for i := 0; i < len(c.targets); i++ {
		select {
		case e := <-doneCh:
			if e != nil {
				return e
			}
		case <-ctx.Done():
			return ctx.Err()
		}
	}
	
	return nil
```

Reply:
I'm not convinced. I think that even when this form is simpler, it somehow obscures when the `pending` is decreased because the select also has the `ctx.Done()` case.